core/src/miner/miner.rs

```diff
@@ -791,6 +794,8 @@ impl MinerService for Miner {
             // | Make sure to release the locks before calling that method.     |
             // ------------------------------------------------------------------
             self.update_sealing(client);
+        } else if self.parcel_reseal_allowed() && !client.is_timer_set() {
+            client.set_timer(self.options.reseal_max_period);
```

Comment:
I think you should call `client.update_sealing()` or `client.set_timer(0);`.

Reply:
IMO, `client.set_timer(0)` will be better.

Reply:
Then why the if statement is needed?

Reply:
This statement is deleted.